pkg/serializer/encode.go

```diff
@@ -1,14 +1,26 @@
 package serializer
 
 import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+
 	"github.com/weaveworks/libgitops/pkg/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/kustomize/kyaml/comments"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
 type EncodingOptions struct {
 	// Use pretty printing when writing to the output. (Default: true)
+	// TODO: Fix that sometimes omitempty fields aren't respected
 	Pretty *bool
+	// Whether to preserve YAML comments internally. Only applicable to ContentTypeYAML framers.
+	// Using any other framer will be silently ignored. Usage of this option also requires setting
+	// the PreserveComments in EncodingOptions, too. (Default: false)
+	PreserveComments *bool
```

Comment:
Similar to decoder, add a note here that this only works for types implementing `metav1.ObjectMeta`